pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -196,12 +204,19 @@ func newReconciler(ofClient openflow.Client, ifaceStore interfacestore.Interface
 		ofClient:          ofClient,
 		ifaceStore:        ifaceStore,
 		lastRealizeds:     sync.Map{},
-		idAllocator:       newIDAllocator(),
+		idAllocator: newIDAllocator(),
 		priorityAssigners: priorityAssigners,
 	}
 	return reconciler
 }
 
+func (r *reconciler) Run(stopCh <-chan struct{}) {
```

Comment:
I prefer to indicate it is to run IDAllocator worker. So, maybe RunIDAllocatorWorker().

Reply:
I suggested "Run" to keep the interface "Reconcile" generic and implementation agnostic for the consideration that it's possible to replace the dataplane implementation but reuse the network policy controller. For now "Reconcile" just has business concepts: CompletedRule, Reconcile, Forget. `GetRuleByFlowID` and `RunIDAllocatorWorker` introduces openflow specific concept to it. But of course we could refactor this later when the requirement comes.
For now, I think ID allocator is implementation specific and shouldn't be called by networkPolicyController, maybe reconciler could use "Run" as the interface to hide the implementation detail and allocator could use startIDAllocatorWorker to be specific which is called by reconciler.Run. Does it make sense? @jianjuns @srikartati

Reply:
Thanking for sharing your view on this. 
If I understand correctly, you think the right way is to define an interface for IDAllocator and pass it to the network policy controller/reconciler. And the IDAllocator should be initialized in cmd/agent/agent.go and start the worker from there. Is that correct?